ext/node/polyfills/_util/std_asserts.ts

```diff
@@ -103,7 +103,7 @@ export function equal(c: unknown, d: unknown): boolean {
       return aTime === bTime;
     }
     if (typeof a === "number" && typeof b === "number") {
-      return NumberIsNaN(a) && NumberIsNaN(b) || a === b;
+      return NumberIsNaN(a) && NumberIsNaN(b) || ObjectIs(a, b);
```

Comment:
`Object.is(NaN, NaN)` is `true`.

```suggestion
      return ObjectIs(a, b);
```

Reply:
Thanks!